### PR TITLE
Base app structure and user authentication endpoint

### PR DESCRIPTION
**Pull request by miroslaw-dubaj**
/user API ready for frontend implementation

### CODE REVIEW

File: back/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const session = require('express-session');
+const { router } = require('./routing/index')
+const app = express();
+
+app.use(bodyParser.json());
+app.use(session({ secret: 'keyboard cat', cookie: { maxAge: 60000 } }))
```
Comment (monikasudol) [inline back/index.js]: missing semicolons and please change secret key

File: back/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+app.use(router)
+
+app.listen(8080, 'localhost', () => console.log('Shuffeling cards... get ready'));
```
Comment (monikasudol) [inline back/index.js]: missing last empty line

File: back/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (monikasudol) [inline back/.gitignore]: missing empty line

File: back/models/user.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+let counter = 1;
+
+class User {
+  constructor(name) {
+    this.userId = counter++;
+    this.name = name;
+  }
+}
+
+module.exports = { User };
```
Comment (monikasudol) [inline back/models/user.js]: missing empty line, please add it :)

File: back/utils/deck.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const createDeck = (figures, colors) => {
+  const deck = [];
+  figures.forEach((figure) => {
+    colors.forEach((color) => {
+      deck.push({
+        figure,
+        color
+      })
+    })
+  })
```
Comment (monikasudol) [inline back/utils/deck.js]: please add semicolons

File: back/.gitignore (hunk 3 above)
Comment (miroslaw-dubaj) [reply back/.gitignore]: done

File: back/index.js (hunk 1 above)
Comment (miroslaw-dubaj) [reply back/index.js]: fixed

File: back/index.js (hunk 2 above)
Comment (miroslaw-dubaj) [reply back/index.js]: done

File: back/models/user.js (hunk 4 above)
Comment (miroslaw-dubaj) [reply back/models/user.js]: done

File: back/utils/deck.js (hunk 5 above)
Comment (miroslaw-dubaj) [reply back/utils/deck.js]: semicolons added, resolved

**Comment by monikasudol** [comment]
Great job! :)

**Pull request merged**